### ISSUE TITLE
[Feature Request]: GSCA (generalized structured component analysis) to the SEM module

### ISSUE THREAD

**Issue by hhwang87** [jasp-stats/jasp-issues#3055]
### Description

I would like to kindly request the addition of GSCA (generalized structured component analysis) to the SEM model. GSCA is a component-based SEM method similar to PLSPM/PLS-SEM, which is already implemented in the module. However, GSCA offers more flexibility in model specification and estimation, often leading to more efficient estimates. An R package for GSCA, called “gesca,” is available (https://cran.r-project.org/web/packages/gesca/index.html) and can be used similarly to the lavaan or cSEM package. Please let me know if you have any questions or if I can provide further assistance. 

### Is your feature request related to a problem?

No

### Is your feature request related to a JASP module?

SEM

**Comment by tomtomme** [comment]
Thx for the fine request. We already discussed the possibility of this feature here: https://github.com/jasp-stats/jasp-issues/issues/2356#issuecomment-1929462691

So this will possibly be implemented at some point.

Since I am no expert on this module a possibly dumb question; you do not mean this:
![grafik](https://github.com/user-attachments/assets/1a97541b-f825-4def-bc76-186de746d356)

**Comment by EJWagenmakers** [comment]
Note that the JASP team is ready to assist in case you should like to contribute code!

**Comment by juliuspfadt** [comment]
@floschuberth what do you think?

**Comment by FloSchuberth** [comment]
In general, I like the idea of having GSCA in JASP. 
In cSEM, we have currently only a rudimentary implementation of GSCA, i.e., we can only estimate the models that can also be estimated by PLS. As Heungsun explained GSCA and its derivatives like GSCAm and IGSCA are more flexible than PLS in terms of cross-loadings and structural model specifications. 

To my knowledge there is currently no complete implementation of GSCA in R available. Yes, there is the gesca package, but this package does not offer implementations of GSCAm and IGSCA. Some time ago, I had contact with Gyeongcheol Cho who is one of the GSCAPro developers (https://www.gscapro.com/general-info, a closed stand alone package) to figure out whether we can collaborate on properly implementing GSCA in cSEM. At that time I was told that he is working on a own R implementation of GSCA. So no collaboration has been established. So far, I haven't seen this R implementation. Next to that, I am also working with another colleague on further implementing GSCA (and GSCAm and IGSCA) in cSEM. However, this still takes some time till we have reached the full flexibility that (I)GSCA(m) offers.

Long story short. If you want to use cSEM, you can only estimate the models that can be estimated by PLS, i.e., we cannot make use of GSCA's full capacity. Moreover, to the best of my knowledge there is no complete R implementation of GSCA and its derivatives available. Therefore, I suggest to wait till there is a proper R implementation available that offers all the features of GSCA. Otherwise, you face the risk to end up with a half-cooked implementation in JASP.

**Comment by juliuspfadt** [comment]
okay, thank. Hm, so I am understanding correctly, in theory we could implement GSCA with the `gesca` R-package? Then this would be a standalone analysis next to PLS-SEM. Only the extensions are out of reach, that is, GSCAm and IGSCA? 
We could also wait until both (basic and extensions) are fully implemented in cSEM?

**Comment by hhwang87** [comment]
Yes, you can implement GSCA into JASP using the gesca R package by adding a GUI. This package functions similarly to the lavaan or cSEM package, so a comparable GUI can be utilized. I believe your team already has excellent experience and expertise in creating such a GUI. So, it would be more efficient for your team to implement GSCA based on the gesca package.

I do appreciate Florian's feedback. As he mentioned, the gesca package is specifically focused on GSCA. However, I'd like to highlight that GSCA itself is a statistical method for component-based SEM, similar to PLS/PLS-SEM, which is implemented in JASP. GSCAm is a statistical method for factor-based SEM (or SEM based on the lavaan package), while IGSCA is a mixture of GSCA and GSCAm. All three methods fall under the umbrella term GSCA-SEM. It would be great to have an R package for GSCA-SEM, but I am uncertain when that might become available.

Furthermore, once a JASP version of GSCA is released, I think it would be straightforward to extend it to GSCA-SEM later, as this extension will likely use the same GUI with adjusted estimation options.

However, I understand if your team prefers to wait for an R package for GSCA-SEM to become available. I defer to your team on this decision.

Best,
Heungsun

**Comment by FloSchuberth** [comment]
@juliuspfadt Yes, you can implement GSCA via gesca R package. However, as far as I remember it uses the lavaan model syntax differently, i.e., =~ does not specify a latent variable, but a certain type of composite. So this needs to be clarified/harmonized. Otherwise, JASP users might be confused. Moreover, you should check whether gesca offers all the post estimation metrics that are usually considered. I am sure Heungsun can help you with that. 
Yes, this would be a standalone analysis next to PLS-SEM. 

Considering the extensions, I would not say they are out of reach. In cSEM, we have also implemented GSCAm and we are working on IGSCA. However, they can currently only be used for models that can also be estimated with PLS-SEM, i.e., models without feedback loops and no cross-loadings. 

Yes, sure waiting till they are fully implemented in cSEM would also be an option. Then you could just add a top-down menu where you can select PLS or GSCA. However, I am not sure how long the implementation will take. So if you want to have a GSCA implementation in JASP in the near future, I suggest to look for alternatives. 

@hhwang87: Yes, in general PLS is a component/composite based approach to SEM. However, in JASP/cSEM if you use the =~ operator,PLSc is used by default. So we followed lavaan, which uses the =~ to specify a common factor and automatically choose the PLS approach which produces consistent estimates, i.e., PLSc. in this case. It would be nice to have sth similar for GSCA...

**Comment by TarandeepKang** [comment]
Hi @FloSchuberth Gyeongcheol Cho Has recently written a Matlab program which looks like it covers all the bases. He has noted on his development page that it will soon be possible to (somehow?) run this in R. https://github.com/PsycheMatrica/GSCA_Prime

Best,

Tarandeep

**Comment by hhwang87** [comment]
Hi Tarandeep,

If you're looking for something similar to Gyeongcheol's MATLAB package, there is already an R package for GSCA: https://cran.r-project.org/web/packages/gesca/index.html. This package offers almost the same features.

Best,
Heungsun
==================================================
Heungsun Hwang, Ph.D.
Professor, Department of Psychology, McGill University
2001 McGill College Avenue, Room 712
Montreal, QC H3A 1G1, Canada
Tel: 514-398-8021
Lab: sites.google.com/view/hwanglab<https://sites.google.com/view/hwanglab>
==================================================

**Comment by FloSchuberth** [comment]
@TarandeepKang: Thanks for letting me know. However, it seems that the Matlab implementation 'only' covers GSCA and not GSCAm and IGSCA. The latter two are extensions of GSCA that produce conistent estimates for latent variable models. Therefore, I agree with @hhwang87 that you could use the gesca package...
